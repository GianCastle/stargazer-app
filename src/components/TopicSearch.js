import React from 'react'

export default function TopicSearch({ state, onCriteriaChangeHandler }) {
  return (
    <div className="container-fluid my-5" data-testid="topic-search">
      <div className="row">
        <div className="col-md-6">
          <label htmlFor='topic' className='h5'>Write you favorite topic below. <strong>Hint: ⚛️</strong></label>
          <input
            data-testid="topic-search__input"
            name='topic'
            value={state}
            onChange={onCriteriaChangeHandler}
            className="form-control" type="search your topic" placeholder="Search your topic" aria-label="Search github topic" />
        </div>
      </div>
    </div>
  )
}
