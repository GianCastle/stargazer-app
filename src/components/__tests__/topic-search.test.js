import { fireEvent, render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import TopicSearch from "../TopicSearch";


describe('header component', () => {
  it('should render', () => {
    const { wrapper, getByTestId } = render(<TopicSearch />);
    const header = getByTestId('topic-search');

    expect(header).toHaveTextContent('Write you favorite topic below');
    expect(wrapper).toMatchSnapshot();
  });

  it('should remove default topic and look for other topic', () => {
    let mockState = "mocked-state"
    const mockOnCriteriaChangeHandler = jest.fn((e) => { mockState = e.target.value });

    const { wrapper, getByTestId } = render(<TopicSearch state={mockState} onCriteriaChangeHandler={mockOnCriteriaChangeHandler} />);
    const header = getByTestId('topic-search');

    expect(wrapper).toMatchSnapshot();

    expect(header).toHaveTextContent('Write you favorite topic below');
    expect(getByTestId('topic-search__input')).toHaveValue(mockState);


    fireEvent.change(getByTestId('topic-search__input'), { target: { value: '' } });
    expect(mockOnCriteriaChangeHandler).toHaveBeenCalled();
    expect(mockState).toBe('');

    fireEvent.change(getByTestId('topic-search__input'), { target: { value: 'react-hooks' } });
    expect(mockState).toBe('react-hooks');
  })
})
