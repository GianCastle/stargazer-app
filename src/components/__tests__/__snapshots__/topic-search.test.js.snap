// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header component should remove default topic and look for other topic 1`] = `undefined`;

exports[`header component should render 1`] = `undefined`;
